### PR TITLE
Added the original request ID to BailmentNotice.  Also added the vers…

### PR DESCRIPTION
**Pull request by tomlandrum**
…ion number to child classes of PeerRequest.

### CODE REVIEW

File: src/core/contract/peer/PeerRequest.cpp [hunk 1]
```diff
@@ -90,10 +90,11 @@ PeerRequest::PeerRequest(
 
 PeerRequest::PeerRequest(
     const ConstNym& nym,
+    const std::uint32_t version,
     const Identifier& recipient,
     const Identifier& server,
     const proto::PeerRequestType& type)
-    : ot_super(nym, PEER_REQUEST_VERSION)
```
Comment (justusranvier) [inline src/core/contract/peer/PeerRequest.cpp]: Now that PEER_REQUEST_VERSION is no longer being used it should be removed from Types.hpp

File: src/api/client/Pair.cpp [hunk 2]
```diff
@@ -439,6 +439,9 @@ void Pair::process_pending_bailment(
         issuer.AddRequest(proto::PEERREQUEST_PENDINGBAILMENT, requestID);
 
     if (added) {
+        const Identifier originalRequest(request.pendingbailment().requestid());
+        issuer.SetUsed(proto::PEERREQUEST_BAILMENT, originalRequest, true);
```
Comment (justusranvier) [inline src/api/client/Pair.cpp]: Should test originalRequest for empty() first before calling issuer.SetUsed

File: src/core/contract/peer/PeerRequest.cpp (hunk 1 above)
Comment (tomlandrum) [reply src/core/contract/peer/PeerRequest.cpp]: PEER_REQUEST_VERSION is still used in PeerObject and PeerReply. Do they also need to be changed to use CURRENT_VERSION?

Comment (justusranvier) [reply src/core/contract/peer/PeerRequest.cpp]: All PeerReply objects should be created as version 4.

PeerObjects should be created as version 5.

**Pull request merged**